datafusion/common/src/scalar.rs

```diff
@@ -5584,6 +5584,87 @@ mod tests {
         }
     }
 
+    #[test]
```

Comment:
That looks good, I checked current impl

```
 select struct(1, 2);
+---------------------------+
| struct(Int64(1),Int64(2)) |
+---------------------------+
| {c0: 1, c1: 2}            |
+---------------------------+
1 row in set. Query took 0.002 seconds.
❯ select struct(null, 2);
+-----------------------+
| struct(NULL,Int64(2)) |
+-----------------------+
| {c0: , c1: 2}         |
+-----------------------+
1 row in set. Query took 0.003 seconds.
❯ select null::struct;
This feature is not implemented: Unsupported SQL type Custom(ObjectName([Ident { value: "struct", quote_style: None }]), [])
```

@crepererum will this fix also address the exception above?

Reply:
I ran the same commands on this branch:

```
❯ select struct(1, 2);
+---------------------------+
| struct(Int64(1),Int64(2)) |
+---------------------------+
| {c0: 1, c1: 2}            |
+---------------------------+
1 row in set. Query took 0.002 seconds.
❯ select struct(null, 2);
+-----------------------+
| struct(NULL,Int64(2)) |
+-----------------------+
| {c0: , c1: 2}         |
+-----------------------+
1 row in set. Query took 0.002 seconds.
❯ select null::struct;
This feature is not implemented: Unsupported SQL type Custom(ObjectName([Ident { value: "struct", quote_style: None }]), [])
```

And the results seem the same